# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteDecoderManagerChild.h
# Commit: 57607f8b6d8e
# Full Hash: 57607f8b6d8e1619b5c488494fd57547944c47b1
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P7: Add Supports messages to PRemoteDecoderManager. r=kamidphish,nika,ipc-reviewers
#   
#   Add a synchronous Supports message to the IPDL PRemoteDecoderManager protocol so
#   decoder modules can query for playback support in the actual process that will
#   attempt to do the decoding.
# ==============================================================================

diff -r 1932f2f34993 -r 57607f8b6d8e dom/media/ipc/RemoteDecoderManagerChild.h
--- a/dom/media/ipc/RemoteDecoderManagerChild.h	Tue Oct 20 23:26:25 2020 +0000
+++ b/dom/media/ipc/RemoteDecoderManagerChild.h	Tue Oct 20 23:26:27 2020 +0000
@@ -30,6 +30,9 @@
   static RemoteDecoderManagerChild* GetSingleton(RemoteDecodeIn aLocation);
 
   // Can be called from any thread.
+  static bool Supports(RemoteDecodeIn aLocation,
+                       const SupportDecoderParams& aParams,
+                       DecoderDoctorDiagnostics* aDiagnostics);
   static already_AddRefed<MediaDataDecoder> CreateAudioDecoder(
       const CreateDecoderParams& aParams);
   static already_AddRefed<MediaDataDecoder> CreateVideoDecoder(